### ISSUE TITLE
Anthropic-specific prompts needed for classification and agent tasks

### ISSUE THREAD

**Issue by dhirenmathur** [potpie-ai/potpie#211]
## Issue Summary
While PR #203 fixed basic Anthropic integration, the current prompts are not optimized for Anthropic models. We need to update classification prompts and agent task descriptions to improve performance with Anthropic models while work continues on dynamic prompt injection (#189).

## Current Status
- Anthropic integration is technically working after PR #203
- Existing prompts designed for other models perform sub-optimally with Anthropic
- Work in progress on dynamic prompt injection based on model name (#189)
- Classification and agent task prompts need immediate updates

## Action Items
- [ ] Review and update classification prompts for Anthropic compatibility
- [ ] Revise agent task descriptions to align with Anthropic's capabilities
- [ ] Test updated prompts with Anthropic models
- [ ] Document Anthropic-specific prompt patterns that work well

## Related Issues/PRs
- #189 Dynamic prompt injection based on model name
- #203 Basic Anthropic integration fix

## Additional Context
This is an interim solution while the more comprehensive dynamic prompt injection system is being developed. Once #189 is completed, these Anthropic-specific prompts can be integrated into the dynamic system.

**Comment by simin75simin** [comment]
@dhirenmathur I'd like to contribute to this issue. Could you assigned this to me? Thanks!

**Comment by dhirenmathur** [comment]
Gladly! Thank you and all the best!

**Comment by dhirenmathur** [comment]
Hey @simin75simin what were your final findings?

**Comment by simin75simin** [comment]
could not test comprehensively right now cause an issue in local repo reads: 'Repo' object has no method 'get_contents'

it seems there are two packages used: gitPython (import git) and pyGithub (import github) and pyGithub has get_contents for production mode, which grabs the repo online. 

trying to install production mode for now. @dhirenmathur

**Comment by dhirenmathur** [comment]
@simin75simin understood, we didn't face this bug when we were testing it that day correct? Did you try to test something different? Could you please provide steps to reproduce and open an issue for it? Thanks!

**Comment by simin75simin** [comment]
it seemed this bug is only present for anthropic. still trying to pin down how exactly it happened. maybe it's some network issue on my local PC that uses a proxy & WSL 2. maybe something else. 

on my PC this error was triggered by going into the postgres docker container and added the preferred AI for the defaultuser to be 'anthropic' and followed readme. @dhirenmathur

**Comment by dhirenmathur** [comment]
If you remove the preferred provider from db it works end to end like we tried previously?

**Comment by simin75simin** [comment]
yes, and if i change it to openai it also works, but if i change it to anthropic it gives UNEXPECTED_EOF_WHILE_READING

**Comment by dhirenmathur** [comment]
And to confirm, you used the API to set the provider in the database? (To make sure that the format is correct in the database)

**Comment by simin75simin** [comment]
no but i can try

also if it's openai it works, like
```sql
UPDATE user_preferences
SET preferences = '{"llm_provider": "openai"}'
WHERE user_id = 'defaultuser';
```

**Comment by simin75simin** [comment]
so i tried using API to set provider in database but same error. i got a 200 from Anthropic but then started getting one UNEXPECTED_EOF_WHILE_READING

**Comment by simin75simin** [comment]
looks like some format conversion with Pydantic with PydanticOutputParser

**Comment by simin75simin** [comment]
https://github.com/potpie-ai/potpie/pull/230 @dhirenmathur

**Comment by simin75simin** [comment]
This can be closed right? @dhirenmathur
